### ISSUE TITLE
Updating cargo causes all future builds to fail until local registry is cleared

### ISSUE THREAD

**Issue by Diggsey** [rust-lang/cargo#1846]
When I updated from rust 1.1 to 1.3 (nightlies) the supplied version of cargo stopped working until I deleted `%USER%\.cargo` entirely.

The first build of any project (after a clean) would show a multitude of errors relating to items being behind a feature gate.
Any further attempts to build would have completely different errors, of the form:
`error: found possibly newer version of crate 'std' which '<crate>' depends on`

The contents of the `.cargo` directory should be versioned so that different cargo/rust versions do not conflict.

**Comment by alexcrichton** [comment]
Can you provide a little more contextual information? This shouldn't be happening, so I'm surprised it is.

* What platform is this on?
* Can you gist some logs?
* Will I be able to reproduce this? If so, how?

**Comment by Diggsey** [comment]
I can't reproduce this anymore unfortunately, but I may have been too hasty to blame it on cargo. This is my best guess at what happened:
- I altered PATH to point to rust 1.3, and similarly changed the RustDT settings
- I restarted cmd and eclipse to get the new environment
- it turns out that `File -> Restart` does not actually restart the eclipse process, and so unknown to me, it failed to get the updated PATH (but it WAS using the new version of cargo, found via the RustDT settings)
- I attempted to build the project with eclipse - this failed. It was using a new cargo with an old rustc - I'm unsure if that is expected to work.
- I attempted to invoke `cargo build` directly using cmd - this also failed, despite using the correct version of both rustc and cargo (I checked by passing --version, something I couldn't do from eclipse)
- All future builds failed until I deleted the .cargo directory, at which point builds via cmd started to work.
- After I figured this much out, I restarted eclipse properly, and it also succeeded.
